app/models/manageiq/providers/cloud_manager/server_group.rb

```diff
@@ -0,0 +1,66 @@
+class ManageIQ::Providers::CloudManager::ServerGroup < ::ServerGroup
+  has_many :vms, dependent => :nullify
+
+  def self.class_by_ems(ext_management_system)
+    ext_management_system&.class_by_ems(:ServerGroup)
+  end
+
+  def self.create_server_group_queue(ems_id, options = {})
+    task_opts = {
+      :action => "creating Server Group for cloud id #{cloud_id}",
+      :ems_id => ems_id
+    }
+
+    queue_opts = {
+      :class_name  => 'ManageIQ::Providers::CloudManager::ServerGroup',
+      :method_name => 'create_server_group',
+      :role        => 'ems_operations',
+      :queue_name  => ext_management_system.queue_name_for_ems_operations,
+      :zone        => ext_management_system.my_zone,
+      :args        => [ext_management_system.id, options]
+    }
+
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def self.create_server_group(ems_id, options)
+    raise ArgumentError, _("ems cannot be nil") if ems_id.nil?
+
+    ext_management_system = ExtManagementSystem.find(ems_id)
+
+    klass = ext_management_system.class_by_ems(:ServerGroup)
+
+    server_group_opts = klass.raw_create_server_group(ext_management_system, options)
+    klass.create(server_group_opts.merge(:resource => ext_management_system))
+  end
+
+  def delete_server_group_queue(ems_id, options)
```

Comment:
You don't need `ems_id` for any of the "instance" methods because it is available from the association, you can simply call `ext_management_system` and it will look it up for you

Reply:
For the queue method you do need `userid` though:
```suggestion
  def delete_server_group_queue(userid)
```